autoload/wiki/page.vim

```diff
@@ -470,9 +472,30 @@ function! s:export(start, end, cfg) abort " {{{1
   let l:fwiki = expand('%:p') . '.tmp'
 
   " Parse wiki page content
+  let l:lines = getline(a:start, a:end)
+" Replace link extensions with .html if desired
+  if a:cfg.link_ext_replace
+    " Regex requires that links end with '.', even when wiki_link_extension is
+    " empty.
+    if g:wiki_link_target_type ==# 'md'
+      let l:link_rx = '\[\([^\\\[\]]\{-}\)\](\([^\(\)\\]\{-}\)\.'
+      let l:link_rx = l:link_rx . g:wiki_link_extension
+      let l:link_rx = l:link_rx . '\(#[^#\(\)\\]\{-}\)\{-})'
+      let l:pattern = '[\1](\2.html\3)'
+    elseif g:wiki_link_target_type ==# 'wiki'
+      let l:link_rx = '\[\[\([^\\\[\]]\{-}\)\.' . g:wiki_link_extension
+      let l:link_rx = l:link_rx . '\(#[^#\\\[\]]\{-}\)'
+      let l:link_rx = l:link_rx . '|\([^\[\]\\]\{-}\)\]\]'
+      let l:pattern = '\[\[\1.html\2|\3\]\]'
+    else
+      echoerr 'Unsupported link type for export replacement'
```

Comment:
I agree with this approach, but I think we should be explicit about the origin. I.e., mention `g:wiki_link_target_type` in the error message.